components/script/canvas_state.rs

```diff
@@ -119,6 +119,8 @@ pub(crate) struct CanvasContextState {
     text_baseline: TextBaseline,
     #[no_trace]
     direction: Direction,
+    // number of clips, used for poping
+    clips: usize,
```

Comment:
It would be better to use rustdoc and fully explain this value:

```suggestion
    /// The number of clips pushed onto the context since this state was saved. When restoring
    /// this same number of clips will be popped to restore state.
    clips_pushed_since_save: usize,
```

Reply:
Added docs, but worded it differently, because each state just stores number of clips pushed onto while the state is active; so when dropping the state we need to pop that many clips.